File: backend/Services/DashboardListItem.cs
using Insightboard.Api.Models.Dashboards;

namespace Insightboard.Api.Services;

public class DashboardListItem
{
    public Guid Id { get; set; }
    public required string FileName { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public required DashboardStatus Status { get; set; }
}

File: frontend/src/models/DashboardListItem.ts
import z from "zod";
import { DashboardStatus } from "./DashboardModel";

export const DashboardListItemSchema = z.object({
  id: z.string(),
  fileName: z.string(),
  createdAt: z.string(),
  status: z.enum(DashboardStatus),
});

export type DashboardListItem = z.infer<typeof DashboardListItemSchema>;
